Sources/SwiftCliCore/InputValidator.swift

```diff
@@ -30,7 +30,9 @@ public class InputValidator {
             }
         }
 
-        let patternMetaData = MetaDataBuilder().buildAllMetaData(stitchArray: patternNestedArray )
+        let expandedNestedArray = patternNestedArray.map { NestedArrayBuilder().expandRow(row: $0) }
+
+        let patternMetaData = MetaDataBuilder().buildAllMetaData(stitchArray: expandedNestedArray)
```

Comment:
Consider refactoring this `inputValidation()` method for this PR. Suggestions:
- Use a verb name
- Split into the "three phases" you mentioned during pairing so that this method's responsibility is just to make sure the phases happen in the right order.

Reply:
Changed the name and abstracted out the error handling! It feels more readable now. I also considered bundling up the error handling with the data transformation that happens with it at the same time (so a function would take `[String]`, run the `EmptyRowError` check, and then transform to `[[String]]`), but that seems to invalidate the single responsibility rule.  What do you think?